python/ray/_private/usage/usage_lib.py

```diff
@@ -700,6 +700,37 @@ def get_cluster_status_to_report(gcs_client) -> ClusterStatusToReport:
         return ClusterStatusToReport()
 
 
+def get_cloud_from_metadata_requests() -> str:
+    import requests
+
+    def cloud_metadata_request(url: str, headers: Optional[Dict[str, str]]) -> bool:
+        try:
+            res = requests.get(url, headers=headers, timeout=1)
+            # The requests may be rejected based on pod configuration but if
+            # it's a machine on the cloud provider it should at least be reachable.
+            if res.status_code != 404:
+                return True
+        except requests.exceptions.ConnectionError:
+            pass
+        return False
+
+    # Make internal metadata requests to all 3 clouds
+    if cloud_metadata_request(
+        "http://metadata.google.internal/computeMetadata/v1",
+        {"Metadata-Flavor": "Google"},
+    ):
+        return "gcp"
+    elif cloud_metadata_request("http://169.254.169.254/latest/meta-data/", None):
+        return "aws"
+    elif cloud_metadata_request(
+        "http://169.254.169.254/metadata/instance?api-version=2021-02-01",
+        {"Metadata": "true"},
+    ):
+        return "azure"
+    else:
+        return "unknown"
```

Comment:
we should wrap all the above checks in an unexpected exception handlers and return "unknown" in those cases (and log the unexpected exception)

Reply:
or catch all exceptions inside `cloud_metadata_request`